mozconfig.macm1sample

```diff
@@ -0,0 +1,14 @@
+. $topsrcdir/browser/config/mozconfig
```

Comment:
My one question with this patch is whether we actually need to fork the config for `m1`. I just applied this locally as my normal config and I'm going to see if it builds. If it does, I'd vote to switch everything to the 11.1 SDK.

Reply:
If we build against the 11.1 SDK, will the browser continue to work on older versions of macOS?

Reply:
Oh that's a good question, I have no idea what the SDK version actually affects.

Reply:
The SDK defines the interfaces exposed by system libraries and the APIs which can be called.  If we compile against a newer version of the SDK but aren't calling any APIs which are newly defined in macOS 11 then things *might* work when running on an older version of the OS, but it would be good to test.  We've had problems with this stuff on linux though (we build the driver and node in a container so that we can control the version of libc we are compiling against), and backwards compatibility is not one of Apple's core values.

Reply:
It looks like everything in the build process uses the `MACOSX_DEPLOYMENT_TARGET` env var: https://searchfox.org/mozilla-central/rev/77f0b36028b2368e342c982ea47609040b399d89/build/build-clang/build-clang.py#777 https://github.com/RecordReplay/gecko-dev/blob/4f36a3ecc6da39da48b8724bc7455255d6119e01/build/build-clang/build-clang.py#L770

https://firefox-source-docs.mozilla.org/widget/cocoa/sdks.html was also useful reading.

Given that, the SDK would I assume just fail to build if it could no longer support the target version, and so upgrading the SDK should be fine?

Reply:
Sounds good, thanks for looking into this.